irohad/execution/impl/command_executor.cpp

```diff
@@ -488,220 +490,481 @@ namespace iroha {
     this->creator_account_id = creator_account_id;
   }
 
-  bool CommandValidator::hasPermissions(
+  CommandResult CommandValidator::hasPermissions(
       const shared_model::interface::AddAssetQuantity &command,
       ametsuchi::WsvQuery &queries,
       const shared_model::interface::types::AccountIdType &creator_account_id) {
+    auto command_name = "AddAssetQuantity";
     // Check if creator has MoneyCreator permission.
     // One can only add to his/her account
     // TODO: 03.02.2018 grimadas IR-935, Separate asset creation for distinct
     // asset types, now: anyone having permission "can_add_asset_qty" can add
     // any asset
-    return creator_account_id == command.accountId()
-        and checkAccountRolePermission(
-                creator_account_id, queries, Role::kAddAssetQty);
+    if (creator_account_id != command.accountId()) {
+      return makeCommandError(
+          "has permission command validation failed: creator account "
```

Comment:
Here and further it would be better to specify what kind of command has failed. It would help to understand for user which of his command has failed and for maintainers to explain what he did wrong if message is not sufficient

For example, you can start message with: AddAssetQuantity command validation failed:

Reply:
Name of the command is memorised in the beginning of each validation function and is being sent as a second parameter to makeCommandError, so it will be processed and appended to the error message "upwards"